### ISSUE TITLE
[Bug] zos_job_submit interprets $

### ISSUE THREAD

**Issue by jthyssenrocket** [ansible-collections/ibm_zos_core#1011]
### Is there an existing issue for this?

- [X] There are no existing issues.

### Are the dependencies a supported version?

- [X] The dependencies are supported.

### IBM Z Open Automation Utilities

v1.2.5

### IBM Enterprise Python

v3.9.x

### IBM z/OS Ansible core Version

v1.6.0

### ansible-version

v2.15.x

### z/OS version

v2.5

### Ansible module

zos_job_submit

### Bug description

Trying to submit a JCL from a dataset containing a $ (dollar) sign.
Submission fails. It looks like the $ sign are being interpreted as shell variables by "jsub". 
I'm unable to escape the values in the ansible code. 

Playbook:
```
- name: Submit JCL from library with dollar sign
  hosts: all
  gather_facts: false
  environment: "{{ environment_vars }}"
  tasks:
  - name: Submit jobs
    ibm.ibm_zos_core.zos_job_submit:
      src: "RSTEST.AOC.$RS01$.ADB1310(IEFBR14)"
      location: "DATA_SET"
      max_rc: 4
      wait_time_s: 120
```

### Playbook verbosity output.

The full traceback is:
  File "/tmp/ansible_ibm.ibm_zos_core.zos_job_submit_payload_8onv1gtd/ansible_ibm.ibm_zos_core.zos_job_submit_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_job_submit.py", line 645, in submit_src_jcl
  File "/rsusr/idz/zoautil/lib/zoautil_py/jobs.py", line 86, in submit
    raise JobSubmitException(f"Unable to submit Job: {response.to_dict()}")
fatal: [rs01]: FAILED! => {
    "changed": false,
    "duration": 0,
    "invocation": {
        "module_args": {
            "encoding": null,
            "from_encoding": "UTF-8",
            "location": "DATA_SET",
            "max_rc": 4,
            "return_output": true,
            "src": "RSTEST.AOC.$RS01$.ADB1310(IEFBR14)",
            "temp_file": null,
            "to_encoding": "IBM-1047",
            "volume": null,
            "wait": false,
            "wait_time_s": 120
        }
    },
    "job_id": null,
    "msg": "The JCL has been submitted RSTEST.AOC.$RS01$.ADB1310(IEFBR14) and no job id was returned within the allocated time of 120 seconds. Consider using  module zos_job_query to poll for a long running jobs or increasing the value for `wait_times_s`.",
    "stderr": "Unable to submit Job: {'rc': 8, 'response_format': 'UTF-8', 'stdout_response': '', 'stderr_response': 'BGYSC4004E Failed to open file RSTEST.AOC.$.ADB1310(IEFBR14) for read.\\n', 'command': 'jsub \"RSTEST.AOC.$RS01$.ADB1310(IEFBR14)\"'}",
    "stderr_lines": [
        "Unable to submit Job: {'rc': 8, 'response_format': 'UTF-8', 'stdout_response': '', 'stderr_response': 'BGYSC4004E Failed to open file RSTEST.AOC.$.ADB1310(IEFBR14) for read.\\n', 'command': 'jsub \"RSTEST.AOC.$RS01$.ADB1310(IEFBR14)\"'}"
    ]
}
1638734 1696423765.72366: no more pending results, returning what we have

### Ansible configuration.

```YAML
N/A
```

### Contents of the inventory

```YAML
N/A
```

### Contents of `group_vars` or `host_vars`

```YAML
N/A
```

**Comment by jthyssenrocket** [comment]
`jsub "RSTEST.AOC.$RS01$.ADB1310(IEFBR14)"`
-->
`BGYSC4004E Failed to open file RSTEST.AOC.$.ADB1310(IEFBR14)`

`jsub 'RSTEST.AOC.$RS01$.ADB1310(IEFBR14)'` would work as the single quotes prevent the shell from interpreting the $ as a variable

**Comment by richp405** [comment]
Related to issue #724 
Should be fixed in the same PR

**Comment by fernandofloresg** [comment]
Can confirm this was fixed by #1375 .
